src/org/zaproxy/zap/extension/websocket/manualsend/ManualWebSocketSendEditorDialog.java

```diff
@@ -261,5 +325,124 @@ public void addSeparator() {
 		public void addEndButton(JButton button) {
 			messagePanel.addOptions(button, HttpPanel.OptionsLocation.END);
 		}
+		
+		private JButton getBtnReopen() {
+			if (btnReopen == null) {
+				btnReopen = new JButton();
+				btnReopen.setIcon(REOPEN_ICON);
+				btnReopen.setEnabled(true);
+				btnReopen.setToolTipText(Constant.messages.getString("websocket.manual_send.btn_reopen_hint"));
+				btnReopen.addActionListener(e -> {
+					isOnConnectingProgress(true);
+					String newSecWebSocketKey = null;
+					
+					if(!wsUiHelper.isAlwaysGenerate()){
+						newSecWebSocketKey = wsUiHelper.getWebSocketKey();
+					}
+					
+					ReEstablishConnection reEstablishConnection = new ReEstablishConnection(newSecWebSocketKey);
+					
+					reEstablishConnection.addPropertyChangeListener(propertyChangeEvent -> {
+						if((propertyChangeEvent.getNewValue() == SwingWorker.StateValue.DONE)){
+							isOnConnectingProgress(false);
+							int channelId1;
+							try {
+								channelId1 = reEstablishConnection.get();
+								if(channelId1 < 0){
+									if(channelId1 == ReEstablishConnection.REQUEST_OUT_OF_SCOPE){
+										View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.out_of_scope"));
+									}else if(channelId1 == ReEstablishConnection.IOEXCEPTION){
+										View.getSingleton().showWarningDialog(Constant.messages.getString("websocket.manual_send.fail.unable_reopen"));
+									}
+								}else{
+									wsUiHelper.setSelectedChannelId(channelId1);
+									wsUiHelper.requestFocusChannelComboBox();
+								}
+							} catch (Exception e1) {
+								LOGGER.warn(e1.getStackTrace(), e1);
```

Comment:
- This method already logs the stacktrace, the first parameter should be the message.
 - The level should be error, this is probably an issue in the code.

Reply:
SwingWorker.get() throws two different types of exceptions,  the InterruptedException which is related with thread interruption and ExecutionException with the computation. ExecutionException shouldn't have happened because is handled by doInBackground. The ExecutionException could be happened at following cases: either interrupt the thread manually or something interrupt the thread. The first case won't have happened because there is no way to interrupt manually (that maybe is a disadvantage. May we want to interrupt the thread manually (ex when the dialog will be closed). In reference to the second case, I don't think it's an error (ex if application closed, we don't want to log an error). I am trying to figure out which is the issue in the code in order to address it.

Reply:
I didn't mean there was an issue with the current code, just saying that if we get to this point then there's likely an issue.

Reply:
(Actually there's an issue (NPE) but that's already being handled in other comment.)